filesystem.md

```diff
@@ -106,3 +107,28 @@ The `Storage` facade may be used to interact with any of your configured disks.
 #### Delete A Directory
 
 	Storage::deleteDirectory($directory);
+
+<a name="custom-filesystems"></a>
+## Custom Filesystems
+If you need to use a filesystem that is not provided out of the box, you may configure one using a service provider. Visit [Flysystem](https://github.com/thephpleague/flysystem#adapters) to view a list of available options.
+
+In order to setup your custom filesystem you call the `extend` method on the `Illuminate\Filesystem\FilesystemManager` class. This class is available to you through the `Illuminate\Contracts\Filesystem\Factory` interface which can be injected into the `boot` method in your service provider.
+
+The first argument of the `extend` method is the name of the disk you will use, while the second argument is a closure that gets passed both the `$app` and `$config` variables.  
+
+From this closure you must return an instance of `League\Flysystem\Filesystem`. Use the details provided with each filesystem to set this up.
+
+> **Note:** The $config variable will already contain the proper values defined in `config/filesystems.php` for the provided disk.
+
+#### Dropbox Example
+
+	use Illuminate\Contracts\Filesystem\Factory as FilesystemFactory;
+	
+	public function boot(FilesystemFactory $filesystem)
+	{
+		$filesystem->extend('dropbox',function($app,$config) {
```

Comment:
Make sure to follow Laravel code styles for the examples. For example, this would be 
```php
$filesystem->extend('dropbox', function($app, $config) {
```
(maybe even brace on a new line? Someone else will have to catch that for me.)

Check out other code samples in the docs to see where Taylor puts braces, new lines, spaces, etc.